main.js

```diff
@@ -0,0 +1,188 @@
+selected meal and if it is empty removes the 'visible' class from the form. 
+If breakfast is selected the visible class will be added to that option only, 
+if lunch is selected the visible class will be added to that option only,
+if dinner is selected the visible class will be added to that option.
+*/
+const mealOption = document.getElementsByName('mealOptionsHolder');
+    if (meal.value == 'breakfast'){
+        const breakfastOptions = getElementById('breakfastOptions');
+        breakfastOptions.classList.remove('hidden');
+        breakfastOptions.classList.add('visible');
+    }
+        else if (meal.value == 'lunch'){
+            const lunchOptions = getElementById('lunchOptions');
+            lunchOptions.classList.remove('hidden');
+            lunchOptions.classList.add('visible');
+        }
+            else if (meal.value == 'dinner') {
+                const dinnerOptions = getElementById('dinnerOptions');
+                dinnerOptions.classList.remove('hidden');
+                dinnerOptions.classList.add('visible');
+            }
+/*These statements assign the elements with id values of firstName and 
+lastName to the variables firstName and lastName, the age element with id 
+age to the age variable, the element with the id firstNameHolder to the
+firstNameHolder variable, the element with the id lastNameHolder to the 
+lastNameHolder variable, the element with id ageHolder to the ageHolder variable, 
+the element with the id emailAddress to the emailAddress variable, 
+and the element with id of mealRating to the variable mealRating.
+*/
+const firstName = document.getElementById('firstName');
+const lastName = document.getElementById('lastName');
+const age = document.getElementById('age');
+const firstNameHolder = document.getElementById('firstNameHolder');
+const lastNameHolder = document.getElementById('lastNameHolder');
+const ageHolder = document.getElementById('ageHolder');
+const emailAddress = document.getElementById('emailAdress');
+const mealRating = document.getElementById('mealRating');
+/*This is the event listener to listen for the blur event on the firstNameHolder
+element. It envokes the setNames function.
+*/
+firstNameHolder.addEventListener('blur', function() {
+    setNames(firstName, lastName);
+});
+/*This is the event listener to listen for the blur event on the lastNameHolder
+element. It envokes the setNames function.
+*/
+lastNameHolder.addEventListener('blur', function() {
+    setNames(firstName, lastName);
+});
+/*This is the setNames function that takes the input on the element and sets 
+the first character to upper case.
+*/
+function setNames(fName, lName){
+    let fName = document.getElementById('firstName');
+    let lName = document.getElementById('lastName');
+    fName.charAt(0).toUpperCase();
+    fName.charAt(fName.length).toLowerCase();
+    lName.charAt(0).toUpperCase();
+    lName.charAt(lName.length).toLowerCase();
+}
```

Comment:
I like how you've pulled this out! I think I'd take it a step further - you don't have to fix both firstName and lastName at once - only one will change at a time. so function setNames(name) where the work of getting and setting the value happens back in the listener would make sense.